idx/memory/memory.go

```diff
@@ -130,6 +144,69 @@ func (m *MemoryIdx) Update(entry idx.Archive) {
 	m.Unlock()
 }
 
+func (m *MemoryIdx) addTags(def *schema.MetricDefinition) {
```

Comment:
func name is a bit confusing. we're indexing tags (or "adding tags to the index") for a metricdefinition that already has tags. so maybe call this `indexTags` or at least at a function comment explaining what this does. thanks :)